dual.go

```diff
@@ -229,9 +230,10 @@ func dvBind(op Op, inputs []*dualValue) (retVal *dualValue, err error) {
 
 	var ret Value
 	if ret, err = op.Do(vals...); err == nil {
-		retVal = dvUnit(ret)
+		return dvUnit(ret), nil
+	} else {
+		return nil, errors.Wrap(err, opDoFail)
```

Comment:
actually, I think the op name won't show up. This is because in some complex operations (say `softmax`, there would be multiple ops in the `softmax` function. If `%+v` is used, it will only show that the error happens in `SoftMax()`